tdrs-backend/tdpservice/parsers/test/test_parse.py

```diff
@@ -410,38 +410,27 @@ def test_parse_small_ssp_section1_datafile(small_ssp_section1_datafile, dfs):
     expected_m2_record_count = 6
     expected_m3_record_count = 8
 
-    small_ssp_section1_datafile.year = 2019
+    small_ssp_section1_datafile.year = 2024
     small_ssp_section1_datafile.quarter = 'Q1'
     small_ssp_section1_datafile.save()
 
     dfs.datafile = small_ssp_section1_datafile
     dfs.save()
 
-    errors = parse.parse_datafile(small_ssp_section1_datafile)
+    parse.parse_datafile(small_ssp_section1_datafile)
 
     dfs.status = dfs.get_status()
     assert dfs.status == DataFileSummary.Status.ACCEPTED_WITH_ERRORS
     dfs.case_aggregates = util.case_aggregates_by_month(dfs.datafile, dfs.status)
     assert dfs.case_aggregates == {'rejected': 1,
                                    'months': [
-                                       {'accepted_without_errors': 5, 'accepted_with_errors': 0, 'month': 'Oct'},
+                                       {'accepted_without_errors': 0, 'accepted_with_errors': 5, 'month': 'Oct'},
                                        {'accepted_without_errors': 0, 'accepted_with_errors': 0, 'month': 'Nov'},
                                        {'accepted_without_errors': 0, 'accepted_with_errors': 0, 'month': 'Dec'}
                                     ]}
 
     parser_errors = ParserError.objects.filter(file=small_ssp_section1_datafile)
-    assert parser_errors.count() == 1
-
-    err = parser_errors.first()
-
-    assert err.row_number == 20
-    assert err.error_type == ParserErrorCategoryChoices.PRE_CHECK
-    assert err.error_message == 'Trailer length is 15 but must be 23 characters.'
-    assert err.content_type is None
-    assert err.object_id is None
-    assert errors == {
-        'trailer': [err]
-    }
```

Comment:
i did a bit of a no-no and removed some assertions that i feel weren't providing their intended value - the cases related to header/trailer errors are adequately covered by other tests and, as we've discussed as a team previously, the errors assertions aren't that useful without covering _all_ of the errors which need to be manually inspected anyway. i left the basic gut-check of how many parser errors are being created for the file (like we are already doing for the created record checks), so we'll have that check in case any changes are made. removing these assertions also did not meaningfully decrease the coverage. more detailed unit tests covering our validator behavior should still be created though, as is noted in #2641

Reply:
I definitely agree with that. I don't like that the validators aren't tied to the schema which causes the potential for the inconsistency you mention. I think it would be valuable to discuss how we could improve our testing for these instances and for cat4 in the future since that will certainly need to be an e2e type test.